builtin/mainmenu/dlg_settings_advanced.lua

```diff
@@ -532,20 +532,31 @@ local checkboxes = {} -- handle checkboxes events
 
 local function create_change_setting_formspec(dialogdata)
 	local setting = settings[selected_setting]
+	local width = 10
 	local height = 5.2
+	local iheight = 3
 	if setting.type == "noise_params_2d" or setting.type == "noise_params_3d" then
 		-- Three flags, checkboxes on 2 columns, with a vertical space of 1/2 unit
 		height = 8.7
 	elseif setting.type == "flags" then
 		-- Checkboxes on 2 columns, with a vertical space of 1/2 unit
 		height = 5.2 + math.ceil(#setting.possible / 2) / 2
+	elseif not (setting.type == "bool" or setting.type == "enum" or
+			setting.type == "file" or setting.type == "filepath" or
+			setting.type == "v3f") then
+		-- Make small to make GUI/HUD scaling, font size settable on Android
+		width = 7
+		height = 4
+		iheight = 2.7
 	end
-	local formspec = "size[10," .. height .. ",true]" ..
-			"button[5," .. height - 0.7 .. ";2,1;btn_done;" .. fgettext("Save") .. "]" ..
-			"button[3," .. height - 0.7 .. ";2,1;btn_cancel;" .. fgettext("Cancel") .. "]" ..
+	local formspec = "size[" .. width .. "," .. height .. ",true]" ..
+			"button[" .. width / 2 + 0.5 .. "," .. height - 0.5 .. ";2,1;btn_done;" ..
+				fgettext("Save") .. "]" ..
+			"button[" .. width / 2 - 2.5 .. "," .. height - 0.5 .. ";2,1;btn_cancel;" ..
+				fgettext("Cancel") .. "]" ..
 			"tablecolumns[color;text]" ..
 			"tableoptions[background=#00000000;highlight=#00000000;border=false]" ..
-			"table[0,0;10,3;info;"
+			"table[0,0;" .. width .. "," .. iheight .. ";info;"
```

Comment:
Could you please make use of textareas here? Otherwise users will again complain when their font is too large to display the description lines.

Reply:
I have to learn how, working on it.

Reply:
Done.